tests/transforms/test_qcut.py

```diff
@@ -3630,45 +3630,79 @@ def test_no_cuts_raises(self):
 class TestCutCircuitExpansion:
     """Test of expansion in the cut_circuit function"""
 
-    def test_no_expansion(self, mocker):
+    @pytest.mark.parametrize(
+        "cut_transform, measurement",
+        [
+            (qcut.cut_circuit, qml.expval(qml.PauliZ(0))),
+            (qcut.cut_circuit_mc, qml.sample(wires=[0])),
+        ],
+    )
+    def test_no_expansion(self, mocker, cut_transform, measurement):
         """Test if no/trivial expansion occurs if WireCut operations are already present in the
         tape"""
         with qml.tape.QuantumTape() as tape:
             qml.RX(0.3, wires=0)
             qml.WireCut(wires=0)
             qml.RY(0.4, wires=0)
-            qml.expval(qml.PauliZ(0))
+            qml.apply(measurement)
+
+        spy = mocker.spy(cut_transform, "expand_fn")
+        shots = 10
+        if tape.measurements[0].return_type is qml.measurements.Sample:
+            cut_transform(tape, shots, device_wires=[0])
+        else:
+            cut_transform(tape, device_wires=[0])
 
-        spy = mocker.spy(qcut.cut_circuit, "expand_fn")
-        qcut.cut_circuit(tape, device_wires=[0])
         spy.assert_called_once()
 
-    def test_expansion(self, mocker):
+    @pytest.mark.parametrize(
+        "cut_transform, measurement",
+        [
+            (qcut.cut_circuit, qml.expval(qml.PauliZ(0))),
+            (qcut.cut_circuit_mc, qml.sample(wires=[0])),
+        ],
+    )
+    def test_expansion(self, mocker, cut_transform, measurement):
         """Test if expansion occurs if WireCut operations are present in a nested tape"""
         with qml.tape.QuantumTape() as tape:
             qml.RX(0.3, wires=0)
             with qml.tape.QuantumTape() as _:
                 qml.WireCut(wires=0)
             qml.RY(0.4, wires=0)
-            qml.expval(qml.PauliZ(0))
+            qml.apply(measurement)
 
-        spy = mocker.spy(qcut.cut_circuit, "expand_fn")
-        qcut.cut_circuit(tape, device_wires=[0])
+        spy = mocker.spy(cut_transform, "expand_fn")
+        shots = 10
+        if tape.measurements[0].return_type is qml.measurements.Sample:
+            cut_transform(tape, shots, device_wires=[0])
+        else:
+            cut_transform(tape, device_wires=[0])
 
-        assert spy.call_count == 2
+        assert spy.call_count == 1
```

Comment:
We should still be calling the expansion functionality twice. The issue is that we'll have to have one spy for `cut_transform.expand_fn` and another for `qcut._qcut_expand_fn`, and assert that each is called once.

Before, we had
```
return cut_circuit.expand_fn(tape.expand(), max_depth=max_depth - 1)
```
in [line 1914](https://github.com/PennyLaneAI/pennylane/pull/2399/files#diff-80b23fd1a16334ed3be95fed079f293972ee780704fc083dacc652d3f4c374ddL1914) which meant we could get away with just one spy.

Reply:
This is related to the [discussion we had](https://github.com/PennyLaneAI/pennylane/pull/2340#discussion_r828074890) in the previous PR.